### ISSUE TITLE
Ruff doesn't seem to be detecting redefinitions of functions inside class?

### ISSUE THREAD

**Issue by ragardner** [astral-sh/ruff-vscode#257]
I am not 100% sure but isn't ruff meant to highlight redefinitions of functions outside as well as inside classes?

Inside class - not underlined - not working?
![image](https://github.com/astral-sh/ruff-vscode/assets/26602401/40806fd8-101c-420c-982a-7ea6c2d85838)

```python
class X:
    def testfunc(self):
        x = None
        
    def testfunc(self):
        x = None
```

Outside class - underlined - working
![image](https://github.com/astral-sh/ruff-vscode/assets/26602401/4193c2f7-a8fb-46e7-968d-1108c9723002)
```python
def testfunc():
    x = None
    
def testfunc():
    x = None
```

Only setting I'm running on ruff is line length

ruff version on vscode is latest as of writing

OS: Linux Pop OS latest
Python Version: 3.10.6

**Comment by zanieb** [comment]
Thanks for the report!

I can confirm that `F811` is raised for the function definition but not for the class method definition however flake8 would raise here. It looks like a bug in our implementation — I don't think this is a VSCode related issue though so I'm going to transfer it over to the ruff repository.
